presentation/src/main/java/me/androidbox/presentation/event/viewmodel/EventViewModel.kt

```diff
@@ -123,31 +116,40 @@ class EventViewModel @Inject constructor(
                     )
                 }
             }
-            is EventScreenEvent.OnScheduleAlarmReminder -> {
-                _eventScreenState.update { eventScreenState ->
-                    eventScreenState.copy(
-                        alarmItem = eventScreenEvent.alarmItem
-                    )
-                }
-                setAlarmReminder()
-            }
-            is EventScreenEvent.OnAlarmReminderTextChanged -> {
+            is EventScreenEvent.OnAlarmReminderChanged -> {
                 _eventScreenState.update { eventScreenState ->
                     eventScreenState.copy(
-                        alarmReminderText = eventScreenEvent.textId
+                        alarmReminderText = eventScreenEvent.reminderItem.stringResId,
+                        alarmReminderItem = eventScreenEvent.reminderItem
                     )
                 }
             }
         }
     }
 
-    private fun setAlarmReminder() {
-        alarmScheduler.schedule(eventScreenState.value.alarmItem)
-    }
+    private fun insertEventDetails() {
+        val startDateTime = AlarmReminderProvider.getCombinedDateTime(eventScreenState.value.startTime, eventScreenState.value.startDate)
+        val endDateTime = AlarmReminderProvider.getCombinedDateTime(eventScreenState.value.endTime, eventScreenState.value.endDate)
+        val remindAt = AlarmReminderProvider.getRemindAt(eventScreenState.value.alarmReminderItem, eventScreenState.value.startTime, eventScreenState.value.startDate)
+
+        val event = Event(
+            id = UUID.randomUUID().toString(),
+            title = eventScreenState.value.eventTitle,
+            description = eventScreenState.value.eventDescription,
+            startDateTime = startDateTime.toEpochSecond(),
+            endDateTime = endDateTime.toEpochSecond(),
+            remindAt = remindAt.toEpochSecond(),
+            eventCreatorId = preferenceRepository.retrieveCurrentUserOrNull()?.userId ?: "",
+            isUserEventCreator = false,
+            attendees = "attendee 1", /* TODO Add attendees here */
```

Comment:
I know this is mocked data, but remember attendees is a list, so it's better to have a list